spec/state_of_the_nation_spec.rb

```diff
@@ -245,6 +245,96 @@ class Country < ActiveRecord::Base
     end
   end
 
+  context ".active_in_period" do
+    let(:unbounded_president) { country.presidents.create!(entered_office_at: day(4), left_office_at: nil) }
+    let(:bounded_president) { country.presidents.create!(entered_office_at: day(1), left_office_at: day(4)) }
+    let(:president_with_empty_active_period) { country.presidents.create!(entered_office_at: day(4), left_office_at: day(4)) }
+    let(:president_with_nil_period) { country.presidents.create!(entered_office_at: nil, left_office_at: nil) }
+
+    it "returns true for records that are active in the interval" do
+      expect(bounded_president).to be_active_in_interval(interval_start: day(3), interval_end: day(7))
+      expect(bounded_president).to be_active_in_interval(interval_start: day(3), interval_end: nil)
+      expect(bounded_president).to be_active_in_interval(interval_start: nil, interval_end: day(3))
+      expect(bounded_president).to be_active_in_interval(interval_start: nil, interval_end: nil)
+
+      expect(unbounded_president).to be_active_in_interval(interval_start: day(4), interval_end: day(12))
+      expect(unbounded_president).to be_active_in_interval(interval_start: day(4), interval_end: nil)
+      expect(unbounded_president).to be_active_in_interval(interval_start: nil, interval_end: day(12))
+      expect(unbounded_president).to be_active_in_interval(interval_start: nil, interval_end: nil)
+    end
+
+    it "returns true for records with no start or end activation period set" do
```

Comment:
These tests fail due to the behaviour of state_of_the_nation. Records with a `nil` `start_key` value have a `start` value of `Time.now.utc` which means that they are only deemed true in cases where the interval covers some point in the future. This is a surprising behaviour of SOTN and potentially a bug.

I added these tests to discuss but we should maybe drop them and just track that this doesn't work as an issue on SOTN.

Reply:
I agree with you about the `nil` `start_key` being wrong and I would have thought it would return the equivalent of `Time.at(0).utc.round => 1970-01-01 00:00:00 UTC`.

I would suggest fixing it in this release, bumping a major version and adding a clear note describing the change in behaviour and updating the documentation for the version.